store/tikv/region_cache.go

```diff
@@ -385,6 +385,9 @@ func (c *RegionCache) loadStoreAddr(bo *Backoffer, id uint64) (string, error) {
 	for {
 		store, err := c.pdClient.GetStore(bo.ctx, id)
 		if err != nil {
+			if errors.Cause(err) == goctx.Canceled {
+				return "", errors.Trace(err)
+			}
 			err = errors.Errorf("loadStore from PD failed, id: %d, err: %v", id, err)
 			if err = bo.Backoff(boPDRPC, err); err != nil {
```

Comment:
If the context is canceled, it should not do the retry for every operation related, not only for loading store addresses. I think maybe we could add this error checking into `Backoffer` to make this checking reusable everywhere `Backoffer` is used. The `Backoffer` should add a list of errors for which the retry mechanism should be skipped on initialization:

```
func NewBackoffer(skipRetryErrors []error, ... ) *Backoffer {
		return &Backoffer {
				skipRetryErrors: skipRetryErrors,
				...
		}
} 
```

When `Backoffer.Backoff()` is called,  if any error of `skipRetryErrors` happens, it will skip the retry and return error directly.

```
func (b *Backoffer) Backoff(typ backoffType, err error) error {
		// if err is one of b.skipRetryErrors, return err without retry
		...
```